ocaml/quicktest/quicktest_vm_snapshot.ml

```diff
@@ -0,0 +1,92 @@
+(** Set up snapshot test: create a small VM with a selection of VBDs *)
+let with_setup rpc session_id sr vm_template f =
+  print_endline "Setting up test VM" ;
+  let uuid = Client.Client.VM.get_uuid rpc session_id vm_template in
+  print_endline (Printf.sprintf "Template has uuid: %s%!" uuid) ;
+  Qt.VM.with_new rpc session_id ~template:vm_template (fun vm ->
+      print_endline (Printf.sprintf "Installed new VM") ;
+      print_endline
+        (Printf.sprintf "Using SR: %s"
+           (Client.Client.SR.get_name_label ~rpc ~session_id ~self:sr)
+        ) ;
+      let vdi =
+        Client.Client.VDI.create rpc session_id "small" "description" sr
+          4194304L `user false false [] [] [] []
+      in
+      let vdi2 =
+        Client.Client.VDI.create rpc session_id "small2" "description2" sr
+          4194304L `user false false [] [] [] []
+      in
+      ignore
+        (Client.Client.VBD.create ~rpc ~session_id ~vM:vm ~vDI:vdi
+           ~userdevice:"0" ~bootable:false ~mode:`RW ~_type:`Disk
+           ~unpluggable:true ~empty:false ~other_config:[]
+           ~qos_algorithm_type:"" ~qos_algorithm_params:[] ~device:""
+           ~currently_attached:true
+        ) ;
+      ignore
+        (Client.Client.VBD.create ~rpc ~session_id ~vM:vm ~vDI:vdi2
+           ~userdevice:"1" ~bootable:false ~mode:`RW ~_type:`Disk
+           ~unpluggable:true ~empty:false ~other_config:[]
+           ~qos_algorithm_type:"" ~qos_algorithm_params:[] ~device:""
+           ~currently_attached:true
+        ) ;
+      f rpc session_id vm vdi vdi2
+  )
+
+let test_snapshot rpc session_id vm vdi vdi2 =
+  let snapshot = Client.Client.VM.snapshot rpc session_id vm "Snapshot" [] in
+  let vbds = Client.Client.VM.get_VBDs rpc session_id snapshot in
+  let snap_vbd =
+    List.find
+      (fun vbd -> Client.Client.VBD.get_userdevice rpc session_id vbd = "0")
+      vbds
+  in
+  let snap_vbd2 =
+    List.find
+      (fun vbd -> Client.Client.VBD.get_userdevice rpc session_id vbd = "1")
+      vbds
+  in
+  let snap_vdi = Client.Client.VBD.get_VDI rpc session_id snap_vbd in
+  let snap_vdi2 = Client.Client.VBD.get_VDI rpc session_id snap_vbd2 in
+  let orig_vdi = Client.Client.VDI.get_snapshot_of rpc session_id snap_vdi in
+  let orig_vdi2 = Client.Client.VDI.get_snapshot_of rpc session_id snap_vdi2 in
+  assert (orig_vdi = vdi) ;
+  assert (orig_vdi2 = vdi2)
+
+let test_snapshot_ignore_vdi rpc session_id vm vdi vdi2 =
+  let snapshot =
+    Client.Client.VM.snapshot rpc session_id vm "Snapshot" [vdi2]
+  in
+  let vbds = Client.Client.VM.get_VBDs rpc session_id snapshot in
+  let snap_vbd =
+    List.find
```

Comment:
This could fail; `find_opt` would be safer.

Reply:
If it fails wouldn't the test show a regression?

Reply:
I'm expecting that because Qicktest is using Alcotest and I would expect it to catch exceptions during test execution. But I have not tested this.

Reply:
Is is better like this?
Woth Alcotest.fail